src/steps/google_to_notion.py

```
@@ -0,0 +1,217 @@
+            url_match = re.search(r'https?://[^\s]+notion\.so/[^\s]+', text)
+            if url_match:
+                url = url_match.group(0)
+                # Remove query params
+                clean_url = url.split('?')[0]
+                parts = clean_url.rsplit('-', 1)
+                if len(parts) > 1 and parts[-1]:
+                    potential_id = parts[-1]
+                    # Validate it looks like a hex ID (at least 20 chars)
+                    if len(potential_id) >= 20 and all(c in '0123456789abcdefABCDEF' for c in potential_id):
+                        return potential_id
+    except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)